pkg/dashboard/controller.go

```diff
@@ -391,5 +399,55 @@ func (c *JobController) SetupWithManager(mgr manager.Manager) error {
 			return true
 		},
 	}))
+}
+
+func (c *SecretController) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	secret := &corev1.Secret{}
+	if err := c.cachedReader.Get(ctx, req.NamespacedName, secret); err != nil {
+		mgrLog.Error(err, "get secret failed", "namespacedName", req.NamespacedName)
+		return reconcile.Result{}, nil
+	}
+	if secret.DeletionTimestamp != nil {
+		return reconcile.Result{}, nil
+	}
+	c.secretIndexes.addIndex(
+		secret,
+		func(p *corev1.Secret) metav1.ObjectMeta { return p.ObjectMeta },
+		func(name types.NamespacedName) (*corev1.Secret, error) {
+			var s corev1.Secret
+			err := c.cachedReader.Get(ctx, name, &s)
+			return &s, err
+		},
+	)
+	mgrLog.V(1).Info("secret created", "namespacedName", req.NamespacedName)
+	return reconcile.Result{}, nil
+}
 
+func (c *SecretController) SetupWithManager(mgr manager.Manager) error {
+	ctr, err := controller.New("secret", mgr, controller.Options{Reconciler: c})
+	if err != nil {
+		return err
+	}
+
+	return ctr.Watch(source.Kind(mgr.GetCache(), &corev1.Secret{}, &handler.TypedEnqueueRequestForObject[*corev1.Secret]{}, predicate.TypedFuncs[*corev1.Secret]{
+		CreateFunc: func(event event.TypedCreateEvent[*corev1.Secret]) bool {
+			return true
+		},
```

Comment:
filter it?

- data contains `metaurl` or `token`
- has `juicefs/secrets` label